### PR TITLE
modularize protobuf package

### PR DESCRIPTION
**Pull request by turbocrime**
work related to #1208

mostly internal organization

significant change is this adds a re-export of all service definitions, which simplifies imports anywhere clients are constructed.

### CODE REVIEW

File: packages/protobuf/src/index.ts
Comment (turbocrime) [inline packages/protobuf/src/index.ts:1]: this is a barrel import, but everything re-exported here is also used in `typeRegistry`, the intended main export of the package. the independent components are also re-exported for convenience of consumers who likely want to access the service definitions &c

File: packages/protobuf/src/penumbra-core.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export { QueryService as AppService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/core/app/v1/app_connect';
+
+export { QueryService as AuctionService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/core/component/auction/v1/auction_connect';
```
Comment (turbocrime) [inline packages/protobuf/src/penumbra-core.ts:3]: `DutchAuction` was removed as a bare type as it's included in `AuctionService`

File: packages/protobuf/README.md [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
 import { jsonOptions } from '@penumbra-zone/protobuf';
-
 import { createChannelTransport } from '@penumbra-zone/transport-dom';
-import type { ServiceType } from '@bufbuild/protobuf';
 
-// unsafely get first available provider
-const getPort = () => Object.values(window[Symbol.for('penumbra')])[0].connect();
+// naively get first available provider
+const provider = Object.values(window[Symbol.for('penumbra')])[0];
+void provider.request();
```
Comment (grod220) [inline packages/protobuf/README.md:37]: Shouldn't they use client package for this?

File: packages/query/src/root-querier.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ export class RootQuerier implements RootQuerierInterface {
   readonly tendermint: TendermintQuerier;
   readonly shieldedPool: ShieldedPoolQuerier;
   readonly ibcClient: IbcClientQuerier;
-  readonly staking: StakingQuerier;
+  readonly staking: StakeQuerier;
```
Comment (grod220) [inline packages/query/src/root-querier.ts:19]: Should be renamed too?

```diff
- readonly staking: StakeQuerier;
+ readonly stake: StakeQuerier;
```

File: packages/services/package.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "@penumbra-zone/services",
   "version": "3.2.1",
+  "private": true,
```
Comment (grod220) [inline packages/services/package.json:4]: No need to publish anymore?

**Review by grod220 — APPROVED** [review]

File: .changeset/little-masks-watch.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@penumbra-zone/protobuf': patch
+---
```
[the remark this answers is not in the document]
Comment (turbocrime) [reply .changeset/little-masks-watch.md:3]: the protobuf package change adds new exports, but doesn't change existing exports. so those are a patchbump

after making this changeset file, i updated users of the package to use the new exports. that does need a new changeset with new version bumps

File: packages/protobuf/README.md (hunk 2 above)
Comment (turbocrime) [reply packages/protobuf/README.md:37]: most developers will. this example is demonstrating use of the type registry this package, not the client package.

a section above this in the readme instructs, 

>If you are looking for a Penumbra extension client
>
>You should install `@penumbra-zone/client`. This package is provided for
developers interested in lower-level work or more detailed configuration.

File: .changeset/little-masks-watch.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (grod220) [reply .changeset/little-masks-watch.md:3]: We don't need another PR to change this, but new features in a backward compatible way would be a minor: https://semver.org/

**Pull request merged**